woocommerce/woocommerce
# [Launch Your Store] Dynamically create Coming Soon page content

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Partial for https://github.com/woocommerce/woocommerce/issues/45113

Create the coming soon page content based on the value of `woocommerce_store_pages_only` option. 

<img width="796" alt="image" src="https://github.com/woocommerce/woocommerce/assets/1922453/be89618b-be24-45d1-bb33-1bbf96f2d2b5">

When settings are changed, the content changes as well. If the merchant has edited the page, the content won't change on Settings change.

This PR does not handle the theme aspect of the Coming Soon page, only the content.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Fire up a JN instance and enable the `launch-your-store` feature flag
2. Go to the Settings page and note the `Restrict to store pages only` setting
4. Note If you need to start over run `wp option delete woocommerce_coming_soon_page_id`, delete the Coming soon page, and visit the Homescreen to kick off creation of a new Coming Soon page
5. See the front end of the Coming Soon page. The content should match the design screenshot above. Again, note this PR only handles the page content, not the templates. If you don't have `Twenty twenty-four` enabled, you may see the title in the page. Handling templates (or lack of) will be resolved in a follow up.
6. Change the visibility setting `Restrict to store pages only` toggle. Refresh the coming soon page, its content will change.
7. Try doing the reverse.
8. Now edit the Coming Soon page and save
9. Try changing the `Restrict to store pages only` setting once again. Note this time the content will not change.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

Update unreleased feature
</details>
